hoki/tests/test_spec.py

```diff
@@ -1,12 +1,95 @@
+    assert fp.name == "LICK/LICK.U", "The filter name is incorrectly imported"
+    assert (fp.wave == wl).all(), "The wavelength bins are incorrectly imported"
+    assert (np.isclose(fp.throughput, transmission)).all(),\
+           "The throughput is incorrectly imported"
+
+
+def test_load_bandpass():
+    # Test loading a build in function
+    h = spec.load_bandpass("johnson,v")
+    p = psp.ObsBandpass("johnson,v")
+    assert (h.wave == p.wave).all(),\
+           "The bins are incorrectly created using `load_bandpass`"
+    assert (h.throughput == p.throughput).all(),\
+           "The throughput is incorrect using `load_bandpass`"
+
+    # test loading a custom filter
+    fp = spec.load_bandpass(f"{data_path}/LICK.LICK.U.xml", suppress=True)
+    assert fp.name == "LICK/LICK.U",\
+           "The name is incorrectly imported using `load_bandpass`"
+    assert (fp.wave == wl).all(),\
+           "The wavelength bins are incorrectly imported using `load_bandpass`"
+    assert (np.isclose(fp.throughput,transmission)).all(),\
+           "The throughput bins are incorrectly imported using `load_bandpass`"
+
+
+    # test file not found
+    with pytest.raises(FileNotFoundError):
+        spec.load_bandpass("f")
+
+def test_get_spectrum():
+    # numpy array input
+    sp = spec.get_spectrum(data.WL.values, data["6.0"].values)
+    assert np.isclose(sp.wave, data.WL).all(),\
+           "The wavelengths are incorrectly created"
+    assert np.isclose(sp.flux, data["6.0"]*3.846e26*1e7).all(),\
+           "The flux is incorrectly converted"
+    assert sp.fluxunits.name == "flam",\
+           "The units are incorrectly set"
+
+    # pandas Series input
+    sp = spec.get_spectrum(data.WL, data["6.0"])
+    assert np.isclose(sp.wave, data.WL).all(),\
+           "A pandas Series is not properly loaded as wavelength"
+    assert np.isclose(sp.flux, data["6.0"]*3.846e26*1e7).all(),\
+           "A pandas Series is not properly loaded as flux"
+    assert sp.fluxunits.name == "flam",\
+           "The units are incorrectly set"
+
+
+def test_apply_bandpass():
+    h = spec.load_bandpass("johnson,v")
+    sp = spec.get_spectrum(data.WL, data["6.0"])
+    obs = spec.apply_bandpass(sp, h)
+    assert (obs.binwave == data.WL).all(),\
+           "The wavelength bins are not properly propegated when \
+           applying the bandpass"
+    # Another test is needed for checking if bandpass properly applied?
+    # Or already done in pysynphot.
+
+def test_flux_to_vegamag():
+    d2vega = 7.767 #pc (from doi:10.1088/0004-6256/136/1/452)
+    vega = psp.Vega*4*np.pi*(d2vega*3.0857e18)**2
+    bp = spec.load_bandpass("johnson,v")
+    obs = spec.apply_bandpass(vega, bp)
+    assert np.isclose(spec.flux_to_vegamag(bp, obs, d2vega),0),\
+           "Vega normalisation is not applied correctly"
+
+    # A another test could be performed by calculating the absolute magnitude
+    # of Vega, which should be around 0.582. However, this is dependent on the
```

Comment:
yeah don't - unit tests are their to check your code in its more technical aspects, rather than the science.